extensions/amp-story/1.0/amp-story.css

```diff
@@ -337,8 +337,9 @@ amp-story-page[active]:not(.i-amphtml-layout) amp-video.i-amphtml-poolbound:not(
   display: none !important;
 }
 
-amp-story-grid-layer > * {
-  pointer-events: auto !important;
```

Comment:
This would reset all publisher CSS incorrectly setting pointer-events, removing it is pretty risky and impossible to test :(

Reply:
There is only 2 values of pointer-events: `auto` and `none`. If they manually set it to `auto`, they receive the same behavior as is currently happening (noop). If they manually set it to `none` the clickable elements will override that property and catch clicks as expected (improvement if added to foreground layers that are "covering" clickable elements).

I don't believe there's a scenario where pubs can incorrectly set the pointer events to break the story.

Reply:
This solution does not scale well and is so easy to break. Each developer needs to set its own pointer-events if they want to receive clicks?
You forgot some, introducing bugs already: e.g. `i-amphtml-story-page-play-button` and maybe some others that we're forgetting about

Reply:
Another example: people using `button` elements to trigger flows, e.g. consent for CCPA implementations

Reply:
`i-amphtml-story-page-play-button` is not added to the inside of a grid-layer, therefore is not affected by this at all. Only the elements inside grid-layers are affected (look at the CSS selectors). Amp consent is on a button outside the grid-layer so doesn't get affected. User-created buttons are not allowed inside grid-layers. You should only look at the valid list of components under "amp-story-grid-layer-allowed-descendants".

> This solution does not scale well

The reason why this doesn't scale is because we (AMP devs) take the burden of making it work, instead of relying on creation tools to add the right values. We do the same for many things, and if someone sends a bug we can just fix it (eg: someone says X component is not clickable -> we make a PR with the CSS rule). "[Developer Experience > Ease of Implementation](https://github.com/ampproject/amphtml/blob/main/docs/design-principles.md#user-experience--developer-experience--ease-of-implementation)". The alternatives are worse for DevX or UX:

- add a new template: the template would still need us to force the pointer events on clickable elements, otherwise they would be non-clickable. Otherwise, if we disallow interactive elements inside this template (eg: if we make a new component similar to grid-layer) we'd need to allowlist every single non-interactive component there, which is even harder to scale.
- Let creators set the right pointer events: it can actually break stories with interactive elements because we can't retroactively ask creators to add pointer events

What alternative would you suggest that doesn't put the burden on the creator and scales better?

Reply:
For user using buttons, doesn't amp convert them to a tags ? If not, why not just add button to the css rule ?

The amp framework is restraining enough, we should be able to list exhaustively every possible interactions isn't it ? That way the framework would also prevent not interactive layer to prevent you from clicking, I don't see how this could be a regression as long as the list is exhaustive and tested on every possible interactive amp components.

Reply:
I'm not aware of any buttons that users can add onto stories, the `button` tag is specifically not allowed in stories. What you refer is links (<a> tags) which we replace the functionality so they are always clickable with the change `amp-story-grid-layer a {pointer-events: auto}` in this PR.

Reply:
up

Reply:
Up again, this is still a very big issue for us.